jobs/JEVS_MESOSCALE_STATS

```diff
@@ -71,12 +71,17 @@ if [ $MODELNAME = nam ] || [ $MODELNAME = rap ]; then
   export EVSINspcotlk=${EVSINspcotlk:-$COMIN/prep/cam}
   export DCOMINmrms=${DCOMINmrms:-$DCOMROOT/ldmdata/obs/upperair/mrms}
   export DCOMINnohrsc=${DCOMINnohrsc:-$DCOMROOT}
+  export EVSIN=${COMIN:-$(compath.py $envir/com/$NET/$evs_ver)}
 
   if [ $VERIF_CASE = snowfall ]; then
-      np=$(date +"%H");         nq=$(($np%6));        nr=$(($np-$nq));
+      export VDATE=${VDATE:-$PDYm2}
+      np=$vhr;         nq=$(($np%6));        nr=$(($np-$nq));
       printf -v ns "%02d" $np;        printf -v nt "%02d" $nr;
-      export cyc=$nt;
-      export cycle=t${cyc}z
+      export vhr=$nt;
+      export cycle=t${vhr}z
```

Comment:
@RoshanShrestha-NOAA This line is already written on line 59 of [jobs/JEVS_MESOSCALE_STATS](https://github.com/NOAA-EMC/EVS/pull/324/files#diff-c23679a08fae71ccafff042336841950f020c0abcf3b5954ea02bc4f7d8fd27f), does it need to appear again in the same script?

Reply:
Yes, this part was added to make sure the snowfall runs pick only the intended vhr.

Reply:
@RoshanShrestha-NOAA `export cycle=t${vhr}z` can only be used to as an input to setpdy.sh (see how it is used on lines 59 and 60. If you are using $cycle for some other purpose, you need to call it something else. 

@ShelleyMelchior-NOAA Do you agree? Perhaps only the `export vhr=$nt;` line matters, and this second instance of `export cycle=t${vhr}z` can be removed?

Reply:
Agreed

Reply:
@RoshanShrestha-NOAA Please work with Binbin to get this change (and the requested dev driver changes) into this PR. Thank you very much!

Reply:
Good point. @RoshanShrestha-NOAA , how is cycle used in line 81? Is it used anywhere downstream? If not, then I wonder if that entire if-block is even needed (lines 76-82)? VDATE for snowfall is determined by the cycle set at line 59 and the subsequent PDY values gathered by running `setpdy.sh` on line 60. The np, nq, nr, and nt variables are local to the J-job script. I can re-run snowfall stats w/ this section commented out and see what happens.

Reply:
> Good point. @RoshanShrestha-NOAA , how is cycle used in line 81? Is it used anywhere downstream? If not, then I wonder if that entire if-block is even needed (lines 76-82)? VDATE for snowfall is determined by the cycle set at line 59 and the subsequent PDY values gathered by running `setpdy.sh` on line 60. The np, nq, nr, and nt variables are local to the J-job script. I can re-run snowfall stats w/ this section commented out and see what happens.

It may throw warning emails if vhr are other than 00,06,12,18. If vhr is set properly, we don't need this part.

Reply:
Ok, so this section is needed to properly handle vhr, but we do not need line 81.

Reply:
> Ok, so this section is needed to properly handle vhr, but we do not need line 81.

We need line 81 too, so that the cycle is set for the correct vhr.

Reply:
@RoshanShrestha-NOAA but do you use cycle anywhere downstream? I do not think so. All cycle usage should have been changed to vhour (by Binbin). The only thing we are permitted to use cycle for is to run `setpdy.sh` as up in lines 58-61. If vhr changes from line 59 to line 80, I do not think it will change the result that `setpdy.sh` produces. And we only need `setpdy.sh` results in order to set VDATE and VDATE is already assigned at line 77 for snowfall stats processing. Does that make sense?

Reply:
> @RoshanShrestha-NOAA but do you use cycle anywhere downstream? I do not think so. All cycle usage should have been changed to vhour (by Binbin). The only thing we are permitted to use cycle for is to run `setpdy.sh` as up in lines 58-61. If vhr changes from line 59 to line 80, I do not think it will change the result that `setpdy.sh` produces. And we only need `setpdy.sh` results in order to set VDATE and VDATE is already assigned at line 77 for snowfall stats processing. Does that make sense?

To make it simple, I have moved the lines np=... to export vhr=$nt above right below the line 58. I think, is that OK?

Reply:
export vhr=${vhr:-00}
if [ $MODELNAME = nam ] || [ $MODELNAME = rap ]; then
  if [ $VERIF_CASE = snowfall ]; then
      np=$vhr;  nq=$(($np%6)); nr=$(($np-$nq)); printf -v nt "%02d" $nr;
      export vhr=$nt;
  fi
fi
export cycle=t${vhr}z
setpdy.sh
. $DATA/PDY

Reply:
Sure -- let me look more closely. Where is your path on disk?

Reply:
> Sure -- let me look more closely. Where is your path on disk?

/lfs/h2/emc/vpppg/noscrub/roshan.shrestha/EVS/jobs/JEVS_MESOSCALE_STATS

Reply:
Ok, I like this, @RoshanShrestha-NOAA ! And one extra bit of refinement, can you combine the 2 if-blocks in lines 82-87 to be an "or" test for VERIF_CASE, since they both assign VDATE to PDYm2?

Reply:
> Ok, I like this, @RoshanShrestha-NOAA ! And one extra bit of refinement, can you combine the 2 if-blocks in lines 82-87 to be an "or" test for VERIF_CASE, since they both assign VDATE to PDYm2?

Done

Reply:
This is great, @RoshanShrestha-NOAA . This solves the problem. Please pass this along to Binbin to update the PR and I will test it! Thanks for working through this. ✔️

Reply:
Done and pushed to the feature branch.

Reply:
@RoshanShrestha-NOAA , I do see one thing that needs corrected. Lines 83 and 93 need to be modified to not accept any previous definitions of VDATE. They both just need to be
```
export VDATE=$PDYm2
``` 
otherwise the VDATE will remain PDYm1 and the code breaks.

Reply:
> @RoshanShrestha-NOAA , I do see one thing that needs corrected. Lines 83 and 93 need to be modified to not accept any previous definitions of VDATE. They both just need to be
> 
> ```
> export VDATE=$PDYm2
> ```
> 
> otherwise the VDATE will remain PDYm1 and the code breaks.

Yes, It looks so, however, these are executed by different dev drivers, so, I have added one if statement for grid2obs case, so that export VDATE=${VDATE:-$PDYm2} works for both lines 83 and 93 as intended.